packages/e2e-tests/package.json

```diff
@@ -21,11 +21,10 @@
 		"@wordpress/e2e-test-utils": "file:../e2e-test-utils",
 		"@wordpress/jest-console": "file:../jest-console",
 		"@wordpress/scripts": "file:../scripts",
-		"expect-puppeteer": "^3.2.0",
```

Comment:
Is it because, via https://github.com/facebook/jest/pull/7119 , we can trust it to be provided through the preset from `jest-puppeteer` as introduced by https://github.com/smooth-code/jest-puppeteer/pull/196 ? In which case, we just need to remove the above import?

Reply:
It looks like Jest doesn't merge `setupFIlesAfterEnv` values from preset and the config itself. It overrides instead:
```
$ npm run test-e2e -- --showConfig

> gutenberg@5.1.1 test-e2e /Users/gziolo/PhpstormProjects/gutenberg
> wp-scripts test-e2e --config packages/e2e-tests/jest.config.js "--showConfig"

{
  "configs": [
    {
      "automock": false,
      "browser": false,
      "cache": true,
      "cacheDirectory": "/private/var/folders/8t/0144m_sn3_q5x_cjrfflbkxm0000gn/T/jest_dx",
      "clearMocks": false,
      "coveragePathIgnorePatterns": [
        "/node_modules/"
      ],
      "cwd": "/Users/gziolo/PhpstormProjects/gutenberg",
      "dependencyExtractor": null,
      "detectLeaks": false,
      "detectOpenHandles": false,
      "errorOnDeprecated": false,
      "filter": null,
      "forceCoverageMatch": [],
      "globalSetup": "/Users/gziolo/PhpstormProjects/gutenberg/node_modules/jest-environment-puppeteer/setup.js",
      "globalTeardown": "/Users/gziolo/PhpstormProjects/gutenberg/node_modules/jest-environment-puppeteer/teardown.js",
      "globals": {},
      "haste": {
        "computeSha1": false,
        "providesModuleNodeModules": []
      },
      "moduleDirectories": [
        "node_modules"
      ],
      "moduleFileExtensions": [
        "js",
        "json",
        "jsx",
        "ts",
        "tsx",
        "node"
      ],
      "moduleNameMapper": {},
      "modulePathIgnorePatterns": [],
      "name": "8d30620c7b75cfd5f4f78d9ee46d32f5",
      "prettierPath": "prettier",
      "resetMocks": false,
      "resetModules": false,
      "resolver": null,
      "restoreMocks": false,
      "rootDir": "/Users/gziolo/PhpstormProjects/gutenberg/packages/e2e-tests",
      "roots": [
        "/Users/gziolo/PhpstormProjects/gutenberg/packages/e2e-tests"
      ],
      "runner": "jest-runner",
      "setupFiles": [
        "/Users/gziolo/PhpstormProjects/gutenberg/packages/e2e-tests/config/gutenberg-phase.js"
      ],
      "setupFilesAfterEnv": [
        "/Users/gziolo/PhpstormProjects/gutenberg/packages/e2e-tests/config/setup-test-framework.js"
      ],
      "skipFilter": false,
      "snapshotSerializers": [],
      "testEnvironment": "/Users/gziolo/PhpstormProjects/gutenberg/node_modules/jest-environment-puppeteer/index.js",
      "testEnvironmentOptions": {},
      "testLocationInResults": false,
      "testMatch": [
        "**/__tests__/**/*.js",
        "**/?(*.)(spec|test).js",
        "**/test/*.js"
      ],
      "testPathIgnorePatterns": [
        "/node_modules/"
      ],
      "testRegex": [],
      "testRunner": "/Users/gziolo/PhpstormProjects/gutenberg/node_modules/jest-jasmine2/build/index.js",
      "testURL": "http://localhost",
      "timers": "real",
      "transform": [
        [
          "^.+\\.[jt]sx?$",
          "/Users/gziolo/PhpstormProjects/gutenberg/node_modules/babel-jest/build/index.js"
        ]
      ],
      "transformIgnorePatterns": [
        "node_modules",
        "scripts/config/puppeteer.config.js"
      ],
      "watchPathIgnorePatterns": []
    }
  ],
  "globalConfig": {
    "bail": 0,
    "changedFilesWithAncestor": false,
    "collectCoverage": false,
    "collectCoverageFrom": null,
    "coverageDirectory": "/Users/gziolo/PhpstormProjects/gutenberg/packages/e2e-tests/coverage",
    "coverageReporters": [
      "json",
      "text",
      "lcov",
      "clover"
    ],
    "coverageThreshold": null,
    "detectLeaks": false,
    "detectOpenHandles": false,
    "errorOnDeprecated": false,
    "expand": false,
    "filter": null,
    "globalSetup": "/Users/gziolo/PhpstormProjects/gutenberg/node_modules/jest-environment-puppeteer/setup.js",
    "globalTeardown": "/Users/gziolo/PhpstormProjects/gutenberg/node_modules/jest-environment-puppeteer/teardown.js",
    "listTests": false,
    "maxConcurrency": 5,
    "maxWorkers": 1,
    "noStackTrace": false,
    "nonFlagArgs": [],
    "notify": false,
    "notifyMode": "failure-change",
    "passWithNoTests": false,
    "projects": null,
    "rootDir": "/Users/gziolo/PhpstormProjects/gutenberg/packages/e2e-tests",
    "runTestsByPath": false,
    "skipFilter": false,
    "testFailureExitCode": 1,
    "testPathPattern": "",
    "testResultsProcessor": null,
    "updateSnapshot": "new",
    "useStderr": false,
    "verbose": null,
    "watch": false,
    "watchman": true
  },
  "version": "24.1.0"
}
```

I will add `expect-puppeteer"` in the config explicitly.

Reply:
> Why was this removed? It's still referenced as an import:
> 
> [gutenberg/packages/e2e-tests/config/setup-test-framework.js](https://github.com/WordPress/gutenberg/blob/5cf3bb9e0cb60a77b92ed5ce1d595a9e4fb9fe3b/packages/e2e-tests/config/setup-test-framework.js#L4)
> 
> Line 4 in [5cf3bb9](/WordPress/gutenberg/commit/5cf3bb9e0cb60a77b92ed5ce1d595a9e4fb9fe3b)
> 
>  import 'expect-puppeteer';

I'm still not sure whether we should include it in the `package.json` explicitly given that it is included in `jest-puppeteer`. It probably doesn't harm to do it. What do you think?

Reply:
> I'm still not sure whether we should include it in the `package.json` explicitly given that it is included in `jest-puppeteer`. It probably doesn't harm to do it. What do you think?

This specific reference is a bit unique than a traditional `import` statement, but I'd prefer we err on the side of: If we reference it as a dependency anywhere, always define it as a dependency. I'd love to see some ESLint code enforcement around this too (i.e. if there's an `import` in a package file, make sure the package declares that as a dependency).

Reply:
For the record, it's referenced in the config file and it repeats what was already provided in the preset:
https://github.com/WordPress/gutenberg/blob/88c8cd7830c5ed037a9061a1d6ae2c340a3ce943/packages/e2e-tests/jest.config.js#L8

I think I will add it as a dependency just for consistency. It would be nicer if we could just include additional entries rather than repeat what was already listed in the preset.

Reply:
Addressed in #14212.